esports_data_from_rimble/rimble/rimble.go

```diff
@@ -0,0 +1,238 @@
+package rimble
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+
+	"github.com/samber/lo"
+)
+
+type PlayerResult struct {
+	Kills      int     `json:"kills"`
+	GameNumber int     `json:"game_number"`
+	KAST       float64 `json:"KAST"`
+	Deaths     int     `json:"deaths"`
+	ADR        float64 `json:"ADR"`
+}
+
+type Player struct {
+	Name     string         `json:"name"`
+	ID       string         `json:"id"`
+	Results  []PlayerResult `json:"results"`
+	Username string         `json:"username"`
+}
+
+type TeamResult struct {
+	GameNumber int `json:"game_number"`
+	TotalKills int `json:"totalKills"`
+	RoundsWon  int `json:"rounds_won"`
+}
+
+type Team struct {
+	GamesWon    int          `json:"games_won"`
+	WinResult   int          `json:"win_result"`
+	Players     []Player     `json:"players"`
+	Name        string       `json:"name"`
+	ID          string       `json:"id"`
+	Designation int          `json:"designation"`
+	Results     []TeamResult `json:"results"`
+}
+
+type Game struct {
+	GameNumber int    `json:"game_number"`
+	MapName    string `json:"map_name"`
+}
+
+type Metadata struct {
+	Games []Game `json:"games"`
+}
+
+type MatchData struct {
+	Metadata    Metadata `json:"metadata"`
+	Team1Name   string   `json:"team_1_name"`
+	Team2Name   string   `json:"team_2_name"`
+	Teams       []Team   `json:"teams"`
+	MatchID     string   `json:"matchid"`
+	MatchStatus string   `json:"match_status"`
+}
+
+func MakeMatchDataFromMatchesJSON(matchesJSON []byte) (MatchData, error) {
+	var matches []MatchData
+	err := json.Unmarshal(matchesJSON, &matches)
+	if err != nil {
+		return MatchData{}, fmt.Errorf("error unmarshaling JSON: %w", err)
+	}
+
+	switch len(matches) {
+	case 0:
+		err = fmt.Errorf("found no matches")
+		return MatchData{}, err
+	case 1:
+		break // only one match found, proceed to return it
+	default:
+		err = fmt.Errorf("found multiple matches")
+		return MatchData{}, err
+	}
+
+	return matches[0], nil
+}
+
+func (match MatchData) Winner() (Team, error) {
+	// create a function that returns true if the team is a winner
```

Comment:
This function contains a lot of comments that are just saying what the code is also saying.  Consider updating so that either the comments add something "interesting" or removing the comments.  For example here, it may be more useful to state the non-obvious items such as:

```
// In the Rimble API, the win result takes values {0, 1} where 0 means did not win and 1 means win.
```
Similar comments throughout the file.

Reply:
OK, I'll go through and remove them unless something is unintutive